loader_logic/dag_abstractions.py

```diff
@@ -1,53 +1,88 @@
+from loader_logic.base_loaders import (
+    load_docs_from_loaders,
+    split_documents,
+    create_embeddings_vectorstore,
+    save_vectorstore_to_disk,
+)
 
-class DagChainBaseLoader(ABC):
 
-    def __init__(self, name: str, loader: List[BaseLoader], schedule='daily'):
+class DagChainBaseLoader(ABC):
+    def __init__(self, name: str, loader: List[BaseLoader], schedule="daily"):
         """Initialize with webpage path."""
+
+        if schedule != "daily":
+            raise ValueError("Only daily schedules are supported right now")
+
         self.name = name
         self.loader = loader
         self.schedule = schedule
 
-    def setup_job(self):
-
-        @asset(name=f"{self.name}_load")
-        def load():
-            asset_loader = self.loader
-            return asset_loader
+    def to_assets(self):
+        @asset(
+            group_name=self.name, name=f"{self.name}_raw_documents", compute_kind="http"
+        )
+        def raw_documents():
+            "Load the raw document text from the source"
+            return load_docs_from_loaders(self.loader)
 
-        embedding = AssetsDefinition.from_op(
-            source_loader.configured({"name": f"{self.name}"}, name=f"{self.name}_text"),
-            keys_by_input_name={"loader_list": AssetKey(f"{self.name}_loader")},
-            keys_by_output_name={"result": AssetKey(f"{self.name}_text")},
+        @asset(
+            group_name=self.name,
+            name=f"{self.name}_documents",
+            ins={"raw_documents": AssetIn(f"{self.name}_raw_documents")},
+            compute_kind="langchain",
         )
+        def documents(raw_documents):
+            "Split the documents into chunks that fit in the LLM context window"
+            return split_documents(raw_documents)
 
-        vectorstore = AssetsDefinition.from_op(
-            output_loader.configured({"name": f"{self.name}"}, name=f"{self.name}_vectorstore"),
-            keys_by_input_name={"source_loader": AssetKey(f"{self.name}_text")},
-            keys_by_output_name={"result": AssetKey(f"{self.name}_vectorstore")},
+        @asset(
+            group_name=self.name,
+            name=f"{self.name}_vectorstore",
+            io_manager_key="vectorstore_io_manager",
+            freshness_policy=FreshnessPolicy(
+                maximum_lag_minutes=5, cron_schedule="0 0 * * *"
+            ),
+            ins={"documents": AssetIn(f"{self.name}_documents")},
+            compute_kind="faiss",
         )
+        def vectorstore(documents):
+            "Compute embeddings and create a vector store"
+            return create_embeddings_vectorstore(documents)
 
+        return [raw_documents, documents, vectorstore]
 
-        @job(name=f"{self.name}_job")
-        def load_vectorstore():
-            vectorstore(embedding(load()))
 
-        
-        if self.schedule != 'daily':
-            print('Warning: only daily schedule supported right now')
+class VectorstoreIOManager(IOManager):
```

Comment:
Nice. I was looking at implementing an IOManager next. Could you see a similar one being used but for storing to a vector DB?

Reply:
Yeah, 100%